daemon/containerd/image_exporter.go

```diff
@@ -270,12 +275,44 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 		}),
 	}
 
+	if platform == nil {
+		// Allow variants to be missing if no specific platform is requested.
+		opts = append(opts, containerd.WithSkipMissing())
+	}
```

Comment:
The platform matcher will handle that as the non-matching platforms will be filtered out and won't be passed to the handler.
This is only needed for the `platform == nil` case, because the matcher matches **any** platform, but has a `Less` implementation that **prefers** the native one.